### PR TITLE
Update to webpack 2

### PR DESCRIPTION
**Pull request by pirelenito**
The initial migration work is completed. Missing tasks:

- [ ] Fix Style extract-text and its unit tests
- [ ] Make the project used in the integration tests more complex so that we can verify it better
- [ ] Update `webpack-merge`
- [ ] Update Babel preset so that we can benefit from Webpack native `import` support

### CODE REVIEW

**Comment by pirelenito** [comment]
https://webpack.js.org/guides/migrating/

File: spec/fixtures/project-content/index.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
 import text from './text.txt'
 
-console.log(text)
+const log = () => console.log(text)
+
+log()
```
Comment (xaviervia) [inline spec/fixtures/project-content/index.js]: This diff looks like it has an amazing explanation

File: src/webpack/loaders/font.js [hunk 2]
```diff
@@ -5,26 +5,64 @@ export default {
   configure () {
     return {
       module: {
-        loaders: [
+        rules: [
```
Comment (xaviervia) [inline src/webpack/loaders/font.js]: Breaking changes are scary. I wonder why this name change in Webpack 2

File: src/webpack/loaders/javascript.js [hunk 3]
```diff
@@ -13,29 +13,20 @@ export default {
...
             include: [
               path.join(projectPath, 'src'),
               ...userPaths
             ],
-            loader: 'babel',
-            query: {
-              // enabling it breaks source maps
-              compact: false
+            loader: 'babel-loader',
+            options: {
```
Comment (xaviervia) [inline src/webpack/loaders/javascript.js]: Sure, it's a better name than `query`, but I wonder if this means that the loaders no longer take URL like "query params"

File: src/webpack/loaders/style.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
+import webpack from 'webpack'
 import path from 'path'
 import postCSSModulesValues from 'postcss-modules-values'
-import ExtractTextPlugin from 'extract-text-webpack-plugin'
+// import ExtractTextPlugin from 'extract-text-webpack-plugin'
```
Comment (xaviervia) [inline src/webpack/loaders/style.js]: Will we no longer extract of is just that this was failing?

File: src/webpack/presets/optimize.js [hunk 5]
```diff
@@ -9,6 +10,9 @@ export default {
 
     return {
       plugins: [
+        // Due to an issue in Webpack, its chunkhashes aren't deterministic.
```
Comment (xaviervia) [inline src/webpack/presets/optimize.js]: Is this a new issue?

File: src/webpack/loaders/font.js [hunk 6]
```diff
@@ -5,26 +5,64 @@ export default {
...
           {
             test: fileExtensions.test.WOFF,
-            loader: 'file?name=[name]-[hash].[ext]&mimetype=application/font-woff'
+            use: [
+              {
+                loader: 'file-loader',
```
Comment (xaviervia) [inline src/webpack/loaders/font.js]: This change makes sense. Reduce confusion.

File: spec/fixtures/project-content/index.js (hunk 1 above)
Comment (pirelenito) [inline spec/fixtures/project-content/index.js]: I wanted to "make sure" it was being transpiled by Babel. I'll improve it before merging.

File: src/webpack/loaders/javascript.js (hunk 3 above)
Comment (pirelenito) [inline src/webpack/loaders/javascript.js]: Exactly, query params are dying in Webpack 2

File: src/webpack/presets/optimize.js (hunk 5 above)
Comment (pirelenito) [inline src/webpack/presets/optimize.js]: No, and it might not even be needed anymore, this was just moved from another file.

**Comment by pirelenito** [comment]
Will need to be reworked... Closing for now.

**Pull request closed without merging**